### ISSUE TITLE
Cannot draw Line in Android/iOS

### ISSUE THREAD

**Issue by EdYao** [gwaldron/osgearth#1307]
As the title says. We have successfully build osgEarth on iOS and Android. but still have some problems, such as the line could not be drawn.  We used the same code as the visual studio project as below to draw a line from New York to Tokyo.

`      

        Geometry* path = new LineString();
        path->push_back( osg::Vec3d(-74, 40.714, 0) );   // New York
        path->push_back( osg::Vec3d(139.75, 35.68, 0) ); // Tokyo

        Feature* pathFeature = new Feature(path, geoSRS);
        pathFeature->geoInterp() = GEOINTERP_GREAT_CIRCLE;

        Style pathStyle;
        pathStyle.getOrCreate<LineSymbol>()->stroke()->color() = Color::White;
        pathStyle.getOrCreate<LineSymbol>()->stroke()->width() = 1.0f;
        pathStyle.getOrCreate<LineSymbol>()->stroke()->smooth() = true;
        pathStyle.getOrCreate<LineSymbol>()->tessellationSize() = 75000;
        pathStyle.getOrCreate<PointSymbol>()->size() = 8;
        pathStyle.getOrCreate<PointSymbol>()->fill()->color() = Color::Red;
        pathStyle.getOrCreate<PointSymbol>()->smooth() = true;
        pathStyle.getOrCreate<AltitudeSymbol>()->clamping() = AltitudeSymbol::CLAMP_TO_TERRAIN;
        pathStyle.getOrCreate<AltitudeSymbol>()->technique() = AltitudeSymbol::TECHNIQUE_GPU;
        pathStyle.getOrCreate<RenderSymbol>()->depthOffset()->enabled() = true;

        //OE_INFO << "Path extent = " << pathFeature->getExtent().toString() << std::endl;

        pathNode = new FeatureNode(pathFeature, pathStyle);
        annoGroup->addChild( pathNode );

        LabelNode* label = new LabelNode("Great circle path", labelStyle);
        label->setPosition(GeoPoint(geoSRS,-170, 61.2));
        labelGroup->addChild(label);
`

It will be very appreciated if anyone could help.

**Comment by gwaldron** [comment]
You may need to call this on your top-level Camera StateSet:

```
#include <osgEarth/GLUtils>
...
GLUtils::setGlobalDefaults( stateset );
```

**Comment by EdYao** [comment]
This solved the problem, which has troubled us for a long time. :)

**Comment by EdYao** [comment]
@gwaldron Thanks for showing me the code to make the line 'visible' now. but the line was above the earth surface, looks like the line altitude was really high, didn't clamp to terrain. If I change the
`    pathStyle.getOrCreate<AltitudeSymbol>()->technique() = AltitudeSymbol::TECHNIQUE_GPU;
`
to
`        pathStyle.getOrCreate<AltitudeSymbol>()->technique() = AltitudeSymbol::TECHNIQUE_DRAPE;
`
, the line just disappeared, but the points (on the line) was there, and did clamp to terrain. I tried all the `enum Technique` and `enum Clamping`, and did not work.

There must something I missed.

**Comment by EdYao** [comment]
@gwaldron In case I didn't describe the problem clearly, I upload a screen record to [here](https://drive.google.com/open?id=10hg8qmHPy3O4ACV69enI-T-7VQlk9Y1I)

**Comment by EdYao** [comment]
@gwaldron Both Android and iOS have this "altitude" [problem](https://drive.google.com/open?id=10hg8qmHPy3O4ACV69enI-T-7VQlk9Y1I)  and we just have no idea what we should do.

**Comment by kinsung** [comment]
LineSymbol don't show, even if the added code: GLUtils: : setGlobalDefaults (stateset);

**Comment by EdYao** [comment]
@kinsung 
`pathStyle.getOrCreate<AltitudeSymbol>()->technique() = AltitudeSymbol::TECHNIQUE_GPU; `
 is required

**Comment by kinsung** [comment]
Thank you for your reply. But it still doesn't work. 😅

**Comment by kinsung** [comment]
I work it out.
The line node must be appended to the scene graph as a child node of the mapnode.

**Comment by EdYao** [comment]
@kinsung That's good.
Did you meet the "altitude" problem on iOS/Android?

**Comment by kinsung** [comment]
Sorry, I can not open this link. And can not figure it out.

**Comment by EdYao** [comment]
@kinsung I upload the screen record to [YouTube](https://www.youtube.com/watch?v=f6qfo0CMWxU&feature=youtu.be)， the line was above earth surface like thousands miles.

**Comment by EdYao** [comment]
@gwaldron If I changed the
`pathStyle.getOrCreate<AltitudeSymbol>()->technique() = AltitudeSymbol::TECHNIQUE_GPU;`

to

`pathStyle.getOrCreate<AltitudeSymbol>()->technique() = AltitudeSymbol::TECHNIQUE_DRAPE;`

The Line didn't draw.

**Comment by EdYao** [comment]
Solved after I update OSGEarth to 2.10.1

**Comment by huangbabaer** [comment]
When the line has an elevation to the ground, the line of view will become very thick when switching the angle of view.
